presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -178,9 +179,29 @@ public TestingPrestoServer(boolean coordinator,
             SqlParserOptions parserOptions,
             List<Module> additionalModules)
             throws Exception
+    {
+        this(coordinator, properties, environment, discoveryUri, parserOptions, additionalModules, Optional.empty());
+    }
+
+    public TestingPrestoServer(boolean coordinator,
+            Map<String, String> properties,
+            String environment,
+            URI discoveryUri,
+            SqlParserOptions parserOptions,
+            List<Module> additionalModules,
+            Optional<Path> baseDataDirPath)
+            throws Exception
     {
         this.coordinator = coordinator;
-        baseDataDir = Files.createTempDirectory("PrestoTest");
+
+        if (!baseDataDirPath.isPresent()) {
+            baseDataDir = createTempDirectory("PrestoTest");
```

Comment:
We can extract a method that throws an unchecked exception, and use a method reference:

```
baseDataDir = baseDataDirPath.orElseGet(TestingPrestoServer::tempDirectory);
isBaseDataDirUserSupplied = baseDataDirPath.isPresent();
```

The method will look like:

```
private static Path tempDirectory() {
    try {
        return createTempDirectory("PrestoTest");
    }
    catch (IOException e) {
        throw new UncheckedIOException(e);
    }
}
```

Reply:
Thanks @nezihyigitbasi . This looks more complex than writing if-else. Is this considered a better practice than writing if-else?

Reply:
It's usually idiomatic to avoid if/else with optionals, and use functional primitives instead. This also fits better with our source code.